### PR TITLE
Fix helm chart can not be deployed without ssl

### LINKED ISSUE #55: [BUG][opensearch] helm chart can not be deployed without ssl
**Describe the bug**
The failure code is:
```yaml
config:
  opensearch_dashboards.yml:
    server:
        name: dashboards
        host: 0.0.0.0
    ## Dashboards TLS Config
      ssl:
        enabled: false
        key: /usr/share/opensearch-dashboards/certs/dashboards-key.pem
        certificate: /usr/share/opensearch-dashboards/certs/dashboards-crt.pem
    opensearch.ssl.certificateAuthorities: /usr/share/opensearch-dashboards/certs/dashboards-root-ca.pem
```
Error Msg:
> {"type":"log","@timestamp":"2021-08-26T05:34:48Z","tags":["fatal","root"],"pid":1,"message":"{ Error: ENOENT: no such file or directory, open '/usr/share/opensearch-dashboards/certs/dashboards-key.pem'\n    at Object.openSync (fs.js:443:3)\n    at readFileSync (fs.js:343:35)\n    at readFile (/usr/share/opensearch-dashboards/src/core/server/http/ssl_config.js:181:31)\n    at new SslConfig (/usr/share/opensearch-dashboards/src/core/server/http/ssl_config.js:131:18)\n    at new HttpConfig (/usr/share/opensearch-dashboards/src/core/server/http/http_config.js:175:16)\n    at MapSubscriber.HttpService.config$.pipe [as project] (/usr/share/opensearch-dashboards/src/core/server/http/http_service.js:61:177)\n    at MapSubscriber._next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/operators/map.js:49:35)\n    at MapSubscriber.Subscriber.next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/Subscriber.js:66:18)\n    at CombineLatestSubscriber.notifyNext (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/observable/combineLatest.js:97:34)\n    at InnerSubscriber._next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/InnerSubscriber.js:28:21)\n    at InnerSubscriber.Subscriber.next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/Subscriber.js:66:18)\n    at MapSubscriber._next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/operators/map.js:55:26)\n    at MapSubscriber.Subscriber.next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/Subscriber.js:66:18)\n    at DistinctUntilChangedSubscriber._next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/operators/distinctUntilChanged.js:69:30)\n    at DistinctUntilChangedSubscriber.Subscriber.next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/Subscriber.js:66:18)\n    at MapSubscriber._next (/usr/share/opensearch-dashboards/node_modules/rxjs/internal/operators/map.js:55:26)\n  errno: -2,\n  syscall: 'open',\n  code: 'ENOENT',\n  path: '/usr/share/opensearch-dashboards/certs/dashboards-key.pem' }"}
> FATAL  Error: ENOENT: no such file or directory, open '/usr/share/opensearch-dashboards/certs/dashboards-key.pem'

**To Reproduce**
Just try to install latest opensearch helm chart with these different configurations:
**Code config 1:**
```yaml
config:
  opensearch_dashboards.yml:
    server:
      name: dashboards
      host: 0.0.0.0
      ## Dashboards TLS Config
    #   ssl:
    #     enabled: false
    #     key: /usr/share/opensearch-dashboards/certs/dashboards-key.pem
    #     certificate: /usr/share/opensearch-dashboards/certs/dashboards-crt.pem
    # opensearch.ssl.certificateAuthorities: /usr/share/opensearch-dashboards/certs/dashboards-root-ca.pem
```

**Code config 2:**
```yaml
config:
  opensearch_dashboards.yml:
    server:
      name: dashboards
      host: 0.0.0.0
      ## Dashboards TLS Config
      ssl:
        enabled: false
    #     key: /usr/share/opensearch-dashboards/certs/dashboards-key.pem
    #     certificate: /usr/share/opensearch-dashboards/certs/dashboards-crt.pem
    # opensearch.ssl.certificateAuthorities: /usr/share/opensearch-dashboards/certs/dashboards-root-ca.pem
```

**Code config 3:**
```yaml
config:
  opensearch_dashboards.yml:
    server:
      name: dashboards
      host: 0.0.0.0
      ## Dashboards TLS Config
      ssl:
        enabled: false
        key: ""
        certificate: "" 
    opensearch.ssl.certificateAuthorities: ""
```

**Expected behavior**
Should be able to install opensearch on Kubernetes Cluster

**Component**
Please list all components currently enabled. (Helm, etc.)

**Screenshots**
None

**Host/Environment (please complete the following information):**
 - OS: Linux -> Kubernetes Cluster with Flatcar OS
 - Version 1.0.0
 - Helm version: 3.5.4

**Additional context**
[Comment under the commit **220a62f** with further error messages.](https://github.com/opensearch-project/opensearch-devops/commit/220a62fa3bc0122872c680eaf18021c589c92c82#commitcomment-55451898)

### PR DESCRIPTION
**Pull request by hgoscenski-imanage**
After switching the name of the config file, and removing the shadowing
between the default (from the docker container opensearch-dashbaords.yaml) and the default from the helm chart (dashboards.yaml) there is an issue with the certs that are attempting to be used.

In order for this to work with the defaults, disabled TLS verification
will be needed, and then disabling TLS to remain in line with the
defaults.

I added a commented out section showing what could potentially be used
as TLS config if the user chooses to enable it.

Signed-off-by: Harrison Goscenski <harrison.goscenski@imanage.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
#55
 
### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by TheAlgo** [comment]
@DandyDeveloper @smlx Please review when you have some time.  After this we should be good to migrate to the new repo for Helm Charts.

**Review by DandyDeveloper — APPROVED** [review]

**Comment by DandyDeveloper** [comment]
Looks good to me. @TheAlgo When this is migrated. Let me know, I probably have a bit of time to work on some stuff today

**Comment by TheAlgo** [comment]
> Looks good to me. @TheAlgo When this is migrated. Let me know, I probably have a bit of time to work on some stuff today

@DandyDeveloper Mostly I will be raising PR for migration by tomorrow because this PR will get merged by tonight.

**Comment by smlx** [comment]
I'm actually not sure this change is a good idea. With `verificationMode: none`, the chart _defaults_ to being insecure, which is just bad.

Could we instead provide a `demo.values.yaml` file which switch off TLS verification if people want to spin up a test cluster?

**Comment by TheAlgo** [comment]
@smlx Did you mean to turn TLS verification on by default and turn it off with a given demo configuration?

**Comment by DandyDeveloper** [comment]
@smlx Insecure yes, but exposed, no. 

Users will and should place an ingress in front of things in order to have some form of TLS from outside of the cluster.

They could also be using a service mesh for internal TLS. 

Unless someone uses an open NodePort / LoadBalancer service to expose the nodes themselves, I'm not exactly sure on why this is a concern. 

"default" values should be indicative of a demo / "bare bones" cluster. At most, we should make users aware that the defaults will leave the cluster in that state. 

Having TLS enabled with the default demo certs has the same headaches and is just as insecure. Demo certificates just serve as a larger headache for the users getting to understand OpenSearch.

That's my two cents.

**Comment by smlx** [comment]
> Did you mean to turn TLS verification on by default and turn it off with a given demo configuration?

Yes that is one option. Another option might be a `insecureDemoConfiguration` boolean value that toggles the demo config (CA certificate etc.).

> At that point, they should be using their own certificates to expose the cluster and configuring the chart as such.

Yes, but disabling verification by default means that Dashboards will just ignore the certificates presented by the Opensearch service and connect anyway, unless the administrator knows to explicitly turn on verification. That seems a bit backwards - insecure configuration should be an explicit choice.

In summary I think that the default configuration here should either be:
* TLS with validation enabled, using the CA certificate from the demo opensearch configuration.
* no TLS at all (if that is possible?).

Having TLS "on" but subtly broken seems like a poor default that will trip up many users who just want to throw the chart into the cluster and have it do something sensible.

**Comment by DandyDeveloper** [comment]
@smlx Agree completely. I'd rather use the demo certificates over no tls if it's a simpler configuration. Which I believe is what we had before?

So, should we actually be fixing the demo certs in dashboard? That seems the right way to be going here?

**Comment by TheAlgo** [comment]
>Yes, but disabling verification by default means that Dashboards will just ignore the certificates presented by the Opensearch service and connect anyway, unless the administrator knows to explicitly turn on verification. That seems a bit backwards - insecure configuration should be an explicit choice.
In summary I think that the default configuration here should either be:
TLS with validation enabled, using the CA certificate from the demo opensearch configuration.
no TLS at all (if that is possible?).
Having TLS "on" but subtly broken seems like a poor default that will trip up many users who just want to throw the chart into the cluster and have it do something sensible.

In the previous charts of ODFE we used to have TLS turned on by default in Elasticsearch and turned off in Kibana. But I agree with you @smlx on the concern you raised regarding poor default. Either we can mention it explicitly in the instructions or as @DandyDeveloper mentioned try to make it like how it was previously and use the demo certs.

**Comment by peterzhuamazon** [comment]
Support demo certificates over no tls at all.

**Comment by TheAlgo** [comment]
@hgoscenski-imanage  Apologies for reiteration of the PR. Will it be possible for you to add the demo certs back to the yaml and turn TLS off by default. TLS is by default turned off for OpenSearch Dashboards by default in TAR and Docker image as well. Thanks

**Comment by TheAlgo** [comment]
In the interest of time for migration. I feel we can work on this particular issue of "poor defaults" as an enhancement because currently the charts are breaking while installation and merge this PR for now. I can create a new issue in the new Helm Chart repo for this.

**Review by peterzhuamazon — APPROVED** [review]
Approve for now @TheAlgo please link the new issue to this PR once created. Thanks.

**Comment by TheAlgo** [comment]
> In the interest of time for migration. I feel we can work on this particular issue of "poor defaults" as an enhancement because currently the charts are breaking while installation and merge this PR for now. I can create a new issue in the new Helm Chart repo for this.

Link for issue : https://github.com/opensearch-project/helm-charts/issues/5
